### PR TITLE
feat: hosted ui for flutter

### PR DESCRIPTION
**Pull request by haverchuck**
*Description of changes:*
Adds Hosted UI docs for amplify-flutter.

**Keeping as draft until code is released.**

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: cspell.json [hunk 1]
```diff
@@ -1157,6 +1157,7 @@
     "ui",
     "UIColor",
     "UINavigationController",
+    "Uint",
```
Comment (jamesonwilliams) [inline cspell.json:1160]: I don't see `Uint` or `webhook` actually getting used in this PR. Can you revert these additions?

File: docs/lib/auth/fragments/flutter/signin_web_ui/20_platform_specific_setup.md [hunk 2]
```diff
@@ -0,0 +1,45 @@
+## How It Works
+Signin with web UI requires the Amplify plugin to display the signin UI inside a webview. This is managed internally by amplify-android and amplify-ios. After the signin process is complete they will redirect back to your app.
```
Comment (Amplifiyer) [inline docs/lib/auth/fragments/flutter/signin_web_ui/20_platform_specific_setup.md:2]: Not sure if we should add the following

>This is managed internally by amplify-android and amplify-ios. After the signin process is complete they will redirect back to your app

File: docs/lib/auth/fragments/flutter/signin_web_ui/10_cli_setup.md [hunk 3]
```diff
@@ -0,0 +1,24 @@
+In terminal, navigate to your project, run `amplify add auth` (or if you've already added it, run `amplify update auth`), and choose the following options:
```
Comment (jamesonwilliams) [inline docs/lib/auth/fragments/flutter/signin_web_ui/10_cli_setup.md:1]:
```suggestion
In the terminal, navigate to your project and run `amplify add auth` (or `amplify update auth`, if you've already added it.) Choose the following options:
```

File: docs/lib/auth/fragments/flutter/signin_web_ui/20_platform_specific_setup.md (hunk 2 above)
Comment (jamesonwilliams) [inline docs/lib/auth/fragments/flutter/signin_web_ui/20_platform_specific_setup.md:2]:
```suggestion
Sign-in with web UI requires the Amplify plugin to display the sign-in UI inside a web view. The UI is managed internally by the Amplify iOS and Android libraries. After the sign-in process is complete, the native libraries will redirect back to your app.
```

File: docs/lib/auth/fragments/flutter/signin_web_ui/20_platform_specific_setup.md [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+
+## Android Platform Setup
+Add the following activity to the `AndroidManifest.xml` file in your app's `Android` directory, replacing `myapp` with whatever value you used for your redirect URI prefix:
```
Comment (jamesonwilliams) [inline docs/lib/auth/fragments/flutter/signin_web_ui/20_platform_specific_setup.md:5]: Should this be lowercase for a directory? `android`?

```suggestion
Add the following activity to the `AndroidManifest.xml` file in your app's `android` directory. Replacing `myapp` with the "redirect signin URI" you provided to the CLI:
```

File: docs/lib/auth/fragments/flutter/signin_web_ui/20_platform_specific_setup.md [hunk 5]
```diff
@@ -0,0 +1,45 @@
...
+          <category android:name="android.intent.category.DEFAULT" />
+          <category android:name="android.intent.category.BROWSABLE" />
+
+          <data android:scheme="myapp" />
+      </intent-filter>
+  </activity>
+  ```
+
+## iOS Platform Setup
+Add the following entry to the URL scheme in the `Info.plist` file in your app's `iOS` directory, replacing `myapp` with whatever value you used for your redirect URI prefix:
```
Comment (jamesonwilliams) [inline docs/lib/auth/fragments/flutter/signin_web_ui/20_platform_specific_setup.md:22]: Same question -- is the directory really stylized with some upper case letters, or is it `ios`?
```suggestion
Add the following entry to the URL scheme in the `Info.plist` file in your app's `iOS` directory. Replace `myapp` with the "redirect signin URI" you provided to the CLI:
```

File: docs/lib/auth/fragments/flutter/signin_web_ui/30_signin.md [hunk 6]
```diff
@@ -0,0 +1,25 @@
+Sweet! You're now ready to launch sign in with web UI.
```
Comment (jamesonwilliams) [inline docs/lib/auth/fragments/flutter/signin_web_ui/30_signin.md:1]: 🍬 
```suggestion
Sweet! You're now ready to launch sign-in with web UI. 
```

File: docs/lib/auth/fragments/flutter/signin_web_ui/30_signin.md [hunk 7]
```diff
@@ -0,0 +1,25 @@
...
+
+```dart
+try {
+  SignInResult res = await Amplify.Auth.signInWithWebUI();
+} on AuthError catch (e) {
+  print(e);
+}
+```
+
+You can also specify a provider with the `provider` attribute.
```
Comment (jamesonwilliams) [inline docs/lib/auth/fragments/flutter/signin_web_ui/30_signin.md:11]:
```suggestion
You can also specify a provider with the `provider` attribute:
```

File: docs/lib/auth/fragments/flutter/signin_web_ui/30_signin.md [hunk 8]
```diff
@@ -0,0 +1,25 @@
...
+
+```dart
+try {
+  SignInResult res = await Amplify.Auth.signInWithWebUI(provider:  AuthProvider.google);
+} on AuthError catch (e) {
+  print(e);
+}
+```
+
+Amplify-Flutter currently supports the following social login providers:
```
Comment (jamesonwilliams) [inline docs/lib/auth/fragments/flutter/signin_web_ui/30_signin.md:21]:
```suggestion
Amplify Flutter currently supports the following social sign-in providers: 
```

**Review by jamesonwilliams — APPROVED** [review]
Miscellaneous minor copy edit suggestions. Nothing major. Looks good.

**Review by Amplifiyer — COMMENTED** [review]
Do we need anything for these pages to show up in the left nav of the docs site?

File: cspell.json (hunk 1 above)
Comment (haverchuck) [reply cspell.json:1160]: The spell check test was failing without this.

File: docs/lib/auth/fragments/flutter/signin_web_ui/10_cli_setup.md (hunk 3 above)
Comment (haverchuck) [reply docs/lib/auth/fragments/flutter/signin_web_ui/10_cli_setup.md:1]: This was taken directly from android docs.

File: docs/lib/auth/fragments/flutter/signin_web_ui/30_signin.md (hunk 6 above)
Comment (haverchuck) [reply docs/lib/auth/fragments/flutter/signin_web_ui/30_signin.md:1]: Taken directly from android docs.

File: docs/lib/auth/fragments/flutter/signin_web_ui/20_platform_specific_setup.md (hunk 4 above)
Comment (haverchuck) [reply docs/lib/auth/fragments/flutter/signin_web_ui/20_platform_specific_setup.md:5]: Pushed a similar revision from my local.

File: docs/lib/auth/fragments/flutter/signin_web_ui/20_platform_specific_setup.md (hunk 5 above)
Comment (haverchuck) [reply docs/lib/auth/fragments/flutter/signin_web_ui/20_platform_specific_setup.md:22]: Pushing similar revision from local

File: docs/lib/auth/fragments/flutter/signin_web_ui/20_platform_specific_setup.md (hunk 2 above)
Comment (haverchuck) [reply docs/lib/auth/fragments/flutter/signin_web_ui/20_platform_specific_setup.md]: image

**Review by Amplifiyer — APPROVED** [review]

File: docs/lib/auth/fragments/flutter/signin_web_ui/20_platform_specific_setup.md (hunk 2 above)
Comment (haverchuck) [reply docs/lib/auth/fragments/flutter/signin_web_ui/20_platform_specific_setup.md]: I edited as we discussed.

**Pull request merged**